### PR TITLE
Add tests for adding/removing/listing repositories.

### PR DESCRIPTION
**Pull request by AMZN-Phil**
Signed-off-by: AMZN-Phil <pconroy@amazon.com>

### CODE REVIEW

File: scripts/o3de/tests/CMakeLists.txt [hunk 1]
```diff
@@ -67,6 +67,13 @@ ly_add_pytest(
     EXCLUDE_TEST_RUN_TARGET_FROM_IDE
 )
 
+ly_add_pytest(
+    NAME o3de_gem_properties
```
Comment (lumberyard-employee-dm) [inline scripts/o3de/tests/CMakeLists.txt:71]: The name for these test are incorrect

**Review by lumberyard-employee-dm — APPROVED** [review]

File: scripts/o3de/tests/test_repo.py [hunk 2]
```diff
@@ -0,0 +1,205 @@
...
+from o3de import manifest, register, repo
+
+TEST_O3DE_MANIFEST_JSON_PAYLOAD = '''
+{
+    "o3de_manifest_name": "testuser",
+    "origin": "C:/Users/testuser/.o3de",
+    "default_engines_folder": "C:/Users/testuser/.o3de/Engines",
+    "default_projects_folder": "C:/Users/testuser/.o3de/Projects",
+    "default_gems_folder": "C:/Users/testuser/.o3de/Gems",
+    "default_templates_folder": "C:/Users/testuser/.o3de/Templates",
```
Comment (AMZN-alexpete) [inline scripts/o3de/tests/test_repo.py:27]: nit: the default folder  for projects, gems and templates should be in `O3DE` not `.o3de` per 
 https://github.com/o3de/o3de/blob/development/scripts/o3de/o3de/manifest.py#L40

File: scripts/o3de/tests/test_repo.py [hunk 3]
```diff
@@ -0,0 +1,205 @@
...
+        def mocked_requests_get(url):
+            if url in self.valid_urls:
+                cm = MagicMock()
+                cm.getcode.return_value = 200
+                cm.read.return_value = 0
+                cm.__enter__.return_value = cm
+            else:
+                raise urllib.error.HTTPError(url, 404, "Not found", {}, 0)
+
+            return cm
```
Comment (AMZN-alexpete) [inline scripts/o3de/tests/test_repo.py:152]: the variable name `cm` does not follow our best practices for variable names - in this case I'm guessing it means 'custom mock?'

Comment (AMZN-Phil) [reply scripts/o3de/tests/test_repo.py:152]: Yes, will update

**Review by AMZN-alexpete — APPROVED** [review]

**Pull request merged**